plugins/lsp-signature: init

wadsaek opened the pull request:
Added a new plugin!

wadsaek commented on plugins/by-name/lsp_signature/default.nix:
```diff
@@ -0,0 +1,206 @@
+    '';
+
+    log_path =
+      defaultNullOpts.mkStrLuaOr lib.types.path # lua
```
thanks, will fix

MattSturgeon commented on tests/test-sources/plugins/by-name/lsp_signature/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+      settings = {
+        debug = true;
+        log_path = /home/wadsaek/.config/lsp_signature.log;
```
When you use an unquoted path like this, nix actually copies it into the nix-store before evaluating the flake. It also complains if the path is outside of the flake's tree:

```
… while evaluating definitions from `/nix/store/XXX-source/plugins/by-name/lsp_signature':
… while evaluating an attribute `log_path`
error: access to absolute path '/home' is forbidden in pure evaluation mode
```

Ideally, this should be set to something that can actually be used when running the test-suite.

OTTOMH, I guess it should be possible to write to `/tmp/*`? But you'd have to pass that as a string to avoid the eval error.

wadsaek replied:
if it's passed as a string, then any writeable path would work i think, but yeah i totally forgot that nix copies all the paths to the store, thanks for reminding

GaetanLepage commented on plugins/by-name/lsp_signature/default.nix:
```diff
@@ -0,0 +1,206 @@
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "lsp_signature";
```
```suggestion
  name = "lsp-signature";
```
Both the module and source files should be named `lsp-signatures`.

wadsaek commented on plugins/by-name/lsp-signature/default.nix:
```diff
@@ -0,0 +1,208 @@
+
+  maintainers = [ lib.maintainers.wadsaek ];
+
+  # Optionally, explicitly declare some options. You don't have to.
```
fixed!

GaetanLepage commented on tests/test-sources/plugins/by-name/lsp-signature/default.nix:
```diff
@@ -0,0 +1,33 @@
+{
+  empty = {
+    plugins.lsp-signature.enable = true;
+  };
+
+  example = {
```
If you don't mind, add a `defaults` test case where you set all the options declared in `settingsOptions` to their plugin default values.

wadsaek replied:
sure!

GaetanLepage commented on plugins/by-name/lsp-signature/default.nix:
```diff
@@ -0,0 +1,208 @@
+    '';
+
+    select_signature_key = defaultNullOpts.mkLua "nil" ''
+      The keybind for cycling to next signature
```
```suggestion
      The keybind for cycling to next signature.
```
Add a period at the end of each sentence. Sorry to bother you with this, but it helps ensuring consistency across our documentation.

wadsaek replied:
will do

GaetanLepage commented on plugins/by-name/lsp-signature/default.nix:
```diff
@@ -0,0 +1,208 @@
+      The keybind for cycling to next signature.
+    '';
+
+    move_cursor_key = defaultNullOpts.mkLua "nil" ''
+      imap, use nvim_set_current_win to move cursor between current window and floating window. Once moved to floating window, you can use <M-d>, <M-u> to move cursor up and down.
+    '';
+
+    # TODO: figure out the types
+    keymaps = defaultNullOpts.mkListOf lib.types.anything [ ] ''
+      related to move_cursor_key. the keymaps inside floating window with arguements of bufnr. 
+      It can be a function that sets keymaps.
+      <M-d> and <M-u> are default keymaps for moving the cursor up and down.
+    '';
```
```suggestion
    move_cursor_key = defaultNullOpts.mkLua "nil" ''
      imap, use `nvim_set_current_win` to move cursor between current window and floating window.
      Once moved to floating window, you can use `<M-d>`, `<M-u>` to move cursor up and down.
    '';

    # TODO: figure out the types
    keymaps = defaultNullOpts.mkListOf lib.types.anything [ ] ''
      Related to `move_cursor_key`.
      The keymaps inside floating window with arguments of `bufnr`.
      It can be a function that sets keymaps.
      `<M-d>` and `<M-u>` are default keymaps for moving the cursor up and down.
    '';
```

GaetanLepage replied:
- Try to wrap "snippets" in backquotes: `foo`.
- Ensure that all sentences start with a capital letter.
- Usually, we break line after each period (to avoid too-long lines). This does not impact the final render.

wadsaek replied:
i think i fixed everything mentioned

GaetanLepage commented:
Also, one of the `default` test is failing: https://buildbot.nix-community.org/#/builders/2783/builds/1174/steps/1/logs/stdio

GaetanLepage commented:
This looks pretty good to me !
Thanks for addressing all the comments.
Please squash your commits to leave only the following two:
- `maintainers: add wadsaek`
- `plugins/lsp-signature: init`

wadsaek commented:
> This looks pretty good to me ! Thanks for addressing all the comments. Please squash your commits to leave only the following two:
> 
> * `maintainers: add wadsaek`
> * `plugins/lsp-signature: init`

is this okay? i've never done squashing before

GaetanLepage commented:
Yes all good !

MattSturgeon commented on plugins/by-name/lsp-signature/default.nix:
```diff
@@ -0,0 +1,212 @@
+        )
+        ''
+          Log directory for when debug is on.
+          Default is `~/.cache/nvim/lsp_signature.log`.
```
We don't normally need to state the default in the description text, because the docs will render the default and the description right next to each other :slightly_smiling_face:

wadsaek replied:
the plugin default has some lua shenanigans, I thought it'd make sense to keep the comment from the original docs

MattSturgeon reviewed (approved):
Don't see any major issues. LGTM, thanks!

FYI the commits are out of order. If someone were to run the testsuite on e05637215f9634e038462ce9fe8ec60f87e072e2, they would get errors that `wadsaek` doesn't exist in `maintainers`.

Not too important, just something to bear in mind.

wadsaek commented:
> Don't see any major issues. LGTM, thanks!
> 
> FYI the commits are out of order. If someone were to run the testsuite on e05637215f9634e038462ce9fe8ec60f87e072e2, they would get errors that `wadsaek` doesn't exist in `maintainers`.
> 
> Not too important, just something to bear in mind.

will this get merged or do i need to do something else? I'm sorry, it's my first time contributing something serious

GaetanLepage commented:
@mergifyio queue
